sysdig/resource_sysdig_secure_managed_policy.go

```diff
@@ -0,0 +1,307 @@
+package sysdig
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	v2 "github.com/draios/terraform-provider-sysdig/sysdig/internal/client/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceSysdigSecureManagedPolicy() *schema.Resource {
+	timeout := 5 * time.Minute
+	return &schema.Resource{
+		CreateContext: resourceSysdigManagedPolicyCreate,
+		ReadContext:   resourceSysdigManagedPolicyRead,
+		UpdateContext: resourceSysdigManagedPolicyUpdate,
+		DeleteContext: resourceSysdigManagedPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(timeout),
+			Delete: schema.DefaultTimeout(timeout),
+			Update: schema.DefaultTimeout(timeout),
+			Read:   schema.DefaultTimeout(timeout),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          "falco",
+				ValidateDiagFunc: validateDiagFunc(validation.StringInSlice([]string{"falco", "list_matching", "k8s_audit", "aws_cloudtrail", "gcp_auditlog", "azure_platformlogs"}, false)),
```

Comment:
Does it make sense to abstract this validateDiagFunc into model.go as well as it will be used for all policy type resources?